### PR TITLE
[Resource] Add HostGroup

### PR DESCRIPTION
**Pull request by michaelkad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPIHostGroupBasic
--- PASS: TestAccIBMPIHostGroupBasic (172.38s)
PASS

...
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_host_group.go [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_remove: {
+				Description: "A workspace ID to stop sharing the host group with.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_Secondaries: {
+				Description: "List of workspace names to share the host group with.",
```
Comment (ismirlia) [inline ibm/service/power/resource_ibm_pi_host_group.go:74]: I think this description has been updated say workspace IDs in the model. The description looks like is correct in the markdown

File: ibm/service/power/resource_ibm_pi_host_group.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+				Required: true,
+				Type:     schema.TypeList,
+			},
+			Arg_Name: {
+				Description: "Name of the host group to create.",
+				ForceNew:    true,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_remove: {
```
Comment (ismirlia) [inline ibm/service/power/resource_ibm_pi_host_group.go:68]: Arg_Remove

Comment (michaelkad) [reply ibm/service/power/resource_ibm_pi_host_group.go:73]:
```
type HostGroupShareOp struct {

	// List of workspaces to share the host group with
	Add []*Secondary `json:"add"`

	// A workspace ID to stop sharing the host group with
	Remove string `json:"remove,omitempty"`
}
```

**Comment by ismirlia** [comment]
I think there's a still a reference to the deprecated "resource." instead of "retry." in the file.

We are still using the SDK state globals here. In some of the refactors, I started placing some generic state variabels since there's some overlap. And I think some of it also like arbitrary and from something the sb returns? That part's a bit confusing.
```
stateConf := &retry.StateChangeConf{
		Pending:    []string{helpers.PIInstanceDeleting},
		Target:     []string{helpers.PIInstanceNotFound},
		Refresh:    isPIHostDeleteRefreshFunc(client, id),
		Delay:      10 * time.Second,
		MinTimeout: 10 * time.Second,
		Timeout:    timeout,
	}
```

Should there newlines between functions? What's the style guideline for that?

File: website/docs/r/pi_host_group.html.markdown [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    pi_name = "name"
+    pi_secondaries = {
+        name = "name"
+        workspace = "workspace"
+    }
+}
+```
+
+## Notes
```
Comment (ismirlia) [inline website/docs/r/pi_host_group.html.markdown:32]:
```suggestion
### Notes
```

**Review by ismirlia — APPROVED** [review]

**Pull request closed without merging**